tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs

```diff
@@ -45,7 +45,29 @@ public LabServiceApi(KeyVaultSecretsProvider keyVault)
             _keyVault = keyVault;
         }
 
-        private LabResponse GetLabResponseFromAPI(UserQueryParameters query)
+        private LabResponse GetLabResponseFromApi(UserQueryParameters query)
+        {
+            //Fetch user
+            string result = CreateLabQuery(query);
+
+            if (string.IsNullOrWhiteSpace(result))
+            {
+                throw new LabUserNotFoundException(query, "No lab user with specified parameters exists");
+            }
+
+            LabResponse response = JsonConvert.DeserializeObject<LabResponse>(result);
+
+            LabUser user = response.User;
+
+            user = JsonConvert.DeserializeObject<LabUser>(result);
+
+            if (!string.IsNullOrEmpty(user.HomeTenantId) && !string.IsNullOrEmpty(user.HomeUPN))
+                user.InitializeHomeUser();
```

Comment:
Would be nice to refactor this to remove this method. The logic in InitializeHomeUser can be made private and invoked on demand.

Reply:
i might do this in a later PR. i'll sync w/you later.

---
In reply to: [241020215](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/721#discussion_r241020215) [](ancestors = 241020215)